### PR TITLE
[PD-1770] Create date fix for Communication Records

### PR DESCRIPTION
**Pull request by koonsies**
## Issue

ContactRecord.Created_On had been set to auto_now. This caused Created_On to update each time the model was updated, thus losing the actual creation date of the object. 

## Solution

This script does the following:
1. Retrieve all min/max dates for all logs in ContactLogEntry for ContactRecord objects
2. Iterate over above min/max dates.
* If last_modified < created_on, last_modified = created_on
* If last_modified < max date of logs, last_modified = max date of logs
* If created_on > min date of logs, created_on = min date of logs

The script additionally creates a CSV of changes made/potential changes (if save() step is omitted).

I originally wrote this script as a migration, but it can potentially run for ~1hr. and didn't want to have that type of migration floating around. Let me know of any concerns for that.

### CODE REVIEW

**Comment by darrint** [comment]
:+1:

**Comment by aspidites** [comment]
Been open for 6 days. Why the DO NOT MERGE?

File: mypartners/correct_auto_now_dates.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from mypartners.models import ContactRecord
+from django.contrib.contenttypes.models import ContentType
+from django.db import connection
+from django.core.exceptions import ObjectDoesNotExist
+
+
+def fix_dates():
+    time_start = time()
+    count_iterations = 0.0
+    count_cd_gt_lm = 0
```
Comment (aspidites) [inline mypartners/correct_auto_now_dates.py]: These variable names...

**Comment by koonsies** [comment]
This script will only run a few times. Keeping it open for visibility.

**Comment by aspidites** [comment]
Has this been run on QC and Staging, yet?

**Comment by koonsies** [comment]
It has not. I was hoping to run it from home, but I hadn't realized that to connect to the AWS servers I needed a VPN. Is there a quick workaround for this?

I would have someone else run it, but it takes a considerable amount of time to process all of the records.

**Pull request closed without merging**